In [1]:
import os
import pandas as pd

# verify csv files

In [2]:

f1 = 'summary.csv'
f2 = 'selfintersections_7_17_2024.csv'

## read `f1`

In [3]:
df1 = pd.read_csv(f1)

In [4]:
df1.head()

folder surface  count
0        cortexode    pial    107
1        cortexode   white    107
2     cortexode-gt    pial    107
3     cortexode-gt   white    107
4  cortexode-gt-rm    pial    107

In [5]:
sum(df1['count'])

7466

There should be 7466 rows in `f2`

## read `f2`

In [6]:
df2 = pd.read_csv(f2,header=None)

In [7]:
df2

0         1       2      3             4         5  \
0         cortexode  210500.0    69.0  white  arctrdagn005  289555.0   
1         cortexode  202048.0   153.0  white  arctrdagn010  208428.0   
2         cortexode  221444.0    87.0  white  arctrdagn009  205725.0   
3         cortexode  225404.0   174.0  white  arctrdagn009  204420.0   
4         cortexode  211628.0    53.0  white  arctrdagn005  297655.0   
...             ...       ...     ...    ...           ...       ...   
7468   pialnn-gt-rm  323185.0  1063.0   pial  arctrdagn017  214423.0   
7469  freesurfer-rm  337901.0    31.0  white  arctrdagn017  223929.0   
7470   pialnn-gt-rm  334374.0   714.0   pial  arctrdagn004  270332.0   
7471  freesurfer-rm  337914.0  1428.0   pial  arctrdagn017  223929.0   
7472   pialnn-gt-rm  363733.0  1005.0   pial  arctrdagn017  223929.0   

                                            6  
0                  289555.white_converted.stl  
1                  208428.white_converted.stl  
2                  205725.white_converted.stl  
3                  204420.white_converted.stl  
4                  297655.white_converted.stl  
...                                       ...  
7468  214423-pialnn-gt-lh_pial-mw-removed.stl  
7469         223929-cortexode-lh_white-rm.stl  
7470  270332-pialnn-gt-lh_pial-mw-removed.stl  
7471          223929-cortexode-lh_pial-rm.stl  
7472  223929-pialnn-gt-lh_pial-mw-removed.stl  

[7473 rows x 7 columns]

In [8]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7473 entries, 0 to 7472
Data columns (total 7 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   0       7473 non-null   object 
 1   1       7466 non-null   float64
 2   2       7466 non-null   float64
 3   3       7466 non-null   object 
 4   4       7466 non-null   object 
 5   5       7466 non-null   float64
 6   6       7466 non-null   object 
dtypes: float64(3), object(4)
memory usage: 408.8+ KB


           note: it has 7473 rows.
           but it should be 7466 rows.

In [9]:
df2.tail()

0         1       2      3             4         5  \
7468   pialnn-gt-rm  323185.0  1063.0   pial  arctrdagn017  214423.0   
7469  freesurfer-rm  337901.0    31.0  white  arctrdagn017  223929.0   
7470   pialnn-gt-rm  334374.0   714.0   pial  arctrdagn004  270332.0   
7471  freesurfer-rm  337914.0  1428.0   pial  arctrdagn017  223929.0   
7472   pialnn-gt-rm  363733.0  1005.0   pial  arctrdagn017  223929.0   

                                            6  
7468  214423-pialnn-gt-lh_pial-mw-removed.stl  
7469         223929-cortexode-lh_white-rm.stl  
7470  270332-pialnn-gt-lh_pial-mw-removed.stl  
7471          223929-cortexode-lh_pial-rm.stl  
7472  223929-pialnn-gt-lh_pial-mw-removed.stl

In [10]:
df2[df2.isnull().any(axis=1)]

0   1   2    3    4   5    6
2748                                 tl NaN NaN  NaN  NaN NaN  NaN
3449                              e.stl NaN NaN  NaN  NaN NaN  NaN
3972  ox2cortex-gt-transformed-pial.stl NaN NaN  NaN  NaN NaN  NaN
5094           -lh_white-mw-removed.stl NaN NaN  NaN  NaN NaN  NaN
7033                         emoved.stl NaN NaN  NaN  NaN NaN  NaN
7322                               .stl NaN NaN  NaN  NaN NaN  NaN
7326                              d.stl NaN NaN  NaN  NaN NaN  NaN

In [11]:
# show those rows before these rows
rows = [2748, 3449, 3972, 5094, 7033, 7322, 7326]
previous_rows = [row-1 for row in rows] + rows

df2.loc[previous_rows]


0         1       2      3  \
2747             deepcsr-gt-transformed  287228.0     0.0  white   
3448            vox2cortex-no-transform   84028.0    48.0  white   
3971                      vox2cortex-gt  283920.0     6.0  white   
5093                      deepcsr-gt-rm  266332.0   408.0   pial   
7032                      freesurfer-rm  205138.0  1024.0   pial   
7321                      freesurfer-rm  245121.0  1010.0   pial   
7325                      freesurfer-rm  255114.0     0.0  white   
2748                                 tl       NaN     NaN    NaN   
3449                              e.stl       NaN     NaN    NaN   
3972  ox2cortex-gt-transformed-pial.stl       NaN     NaN    NaN   
5094           -lh_white-mw-removed.stl       NaN     NaN    NaN   
7033                         emoved.stl       NaN     NaN    NaN   
7322                               .stl       NaN     NaN    NaN   
7326                              d.stl       NaN     NaN    NaN   

                 4         5                                         6  
2747  arctrdagn014  212823.0   212823-deepcsr-gt-transformed-white.stl  
3448  arctrdagn004  257542.0                      257542_v2c_white.ply  
3971  arctrdagn017  202719.0                       202719_lh_white.ply  
5093  arctrdagn016  210617.0  210617-deepcsr-gt-lh_pial-mw-removed.stl  
7032  arctrdagn010  204420.0           204420-cortexode-lh_pial-rm.stl  
7321  arctrdagn004  280941.0           280941-cortexode-lh_pial-rm.stl  
7325  arctrdagn010  205119.0          205119-cortexode-lh_white-rm.stl  
2748           NaN       NaN                                       NaN  
3449           NaN       NaN                                       NaN  
3972           NaN       NaN                                       NaN  
5094           NaN       NaN                                       NaN  
7033           NaN       NaN                                       NaN  
7322           NaN       NaN                                       NaN  
7326           NaN       NaN                                       NaN

     The rows above the ERROR(NULL) ROWS seem to be correct.
 
     throw away the ERROR ROWS

In [12]:
df2 = df2.dropna().reset_index(drop=True)

In [13]:
df2

0         1       2      3             4         5  \
0         cortexode  210500.0    69.0  white  arctrdagn005  289555.0   
1         cortexode  202048.0   153.0  white  arctrdagn010  208428.0   
2         cortexode  221444.0    87.0  white  arctrdagn009  205725.0   
3         cortexode  225404.0   174.0  white  arctrdagn009  204420.0   
4         cortexode  211628.0    53.0  white  arctrdagn005  297655.0   
...             ...       ...     ...    ...           ...       ...   
7461   pialnn-gt-rm  323185.0  1063.0   pial  arctrdagn017  214423.0   
7462  freesurfer-rm  337901.0    31.0  white  arctrdagn017  223929.0   
7463   pialnn-gt-rm  334374.0   714.0   pial  arctrdagn004  270332.0   
7464  freesurfer-rm  337914.0  1428.0   pial  arctrdagn017  223929.0   
7465   pialnn-gt-rm  363733.0  1005.0   pial  arctrdagn017  223929.0   

                                            6  
0                  289555.white_converted.stl  
1                  208428.white_converted.stl  
2                  205725.white_converted.stl  
3                  204420.white_converted.stl  
4                  297655.white_converted.stl  
...                                       ...  
7461  214423-pialnn-gt-lh_pial-mw-removed.stl  
7462         223929-cortexode-lh_white-rm.stl  
7463  270332-pialnn-gt-lh_pial-mw-removed.stl  
7464          223929-cortexode-lh_pial-rm.stl  
7465  223929-pialnn-gt-lh_pial-mw-removed.stl  

[7466 rows x 7 columns]

In [ ]:
df2.to_c